packages/wrangler/src/index.tsx

```diff
@@ -1035,197 +1036,221 @@ function createCLIParser(argv: string[]) {
-              ); // Ugh, I really don't like this message very much
-            }
-            return {
-              binding,
-              id: preview_id,
-            };
+        // Compute zone info from the `host` and `route` args and config;
+        let host = args.host || config.dev.host;
+        let zoneId: string | undefined;
+
+        if (!args.local) {
+          if (host) {
+            zoneId = await getZoneIdFromHost(host);
           }
-        ),
-        // Use a copy of combinedVars since we're modifying it later
-        vars: getVarsForDev(config),
-        wasm_modules: config.wasm_modules,
-        text_blobs: config.text_blobs,
-        data_blobs: config.data_blobs,
-        durable_objects: config.durable_objects,
-        r2_buckets: config.r2_buckets?.map(
-          ({ binding, preview_bucket_name, bucket_name: _bucket_name }) => {
-            // same idea as kv namespace preview id,
-            // same copy-on-write TODO
-            if (!preview_bucket_name) {
-              throw new Error(
-                `In development, you should use a separate r2 bucket than the one you'd use in production. Please create a new r2 bucket with "wrangler r2 bucket create <name>" and add its name as preview_bucket_name to the r2_buckets "${binding}" in your wrangler.toml`
-              );
+          const routes = args.routes || config.route || config.routes;
+          if (!zoneId && routes) {
+            const firstRoute = Array.isArray(routes) ? routes[0] : routes;
+            const zone = await getZoneForRoute(firstRoute);
+            if (zone) {
+              zoneId = zone.id;
+              host = zone.host;
             }
-            return {
-              binding,
-              bucket_name: preview_bucket_name,
-            };
           }
-        ),
-        services: config.services,
-        unsafe: config.unsafe?.bindings,
-      };
+        }
 
-      // mask anything that was overridden in .dev.vars
-      // so that we don't log potential secrets into the terminal
-      const maskedVars = { ...bindings.vars };
-      for (const key of Object.keys(maskedVars)) {
-        if (maskedVars[key] !== config.vars[key]) {
-          // This means it was overridden in .dev.vars
-          // so let's mask it
-          maskedVars[key] = "(hidden)";
+        const nodeCompat = args.nodeCompat ?? config.node_compat;
+        if (nodeCompat) {
+          logger.warn(
+            "Enabling node.js compatibility mode for built-ins and globals. This is experimental and has serious tradeoffs. Please see https://github.com/ionic-team/rollup-plugin-node-polyfills/ for more details."
+          );
         }
-      }
 
-      // now log all available bindings into the terminal
-      printBindings({
-        ...bindings,
-        vars: maskedVars,
-      });
+        const getBindings = (configParam: Config) => ({
+          kv_namespaces: config.kv_namespaces?.map(
```

Comment:
```suggestion
          kv_namespaces: configParam.kv_namespaces?.map(
```
why does this not read from configParam?

Reply:
From my git diffs, it looks like it just got missed in the function conversion. Thanks for the catch!